### PR TITLE
Refactor THPStorage_resize_

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #119749
* __->__ #119671
* #119647

Moving code around to allow it to be reused in the next PR.

### CODE REVIEW

File: aten/src/ATen/native/Resize.cpp [hunk 1]
```diff
@@ -282,4 +286,61 @@ const Tensor& resize__symint(
...
+    src_tensor.resize_({size_bytes});
+
+    // When using resize_ to replace resize_bytes_xxx, in some cases
+    // the original data_ptr is still returned, which is an inconsistent
+    // behavior when compared to resize_bytes_xxx. For these cases,
+    // an additional memory copy and update for storage are required.
+    if (original_data_ptr == src_tensor.storage().data_ptr().get()) {
+      auto new_tensor = at::empty(src_tensor.sizes(), src_tensor.options());
+      new_tensor.copy_(src_tensor);
+      storage.set_data_ptr_noswap(
```
Comment (Skylion007) [inline aten/src/ATen/native/Resize.cpp:332]: Yikes, how is this a const function?

**Review by Skylion007 — COMMENTED** [review]
Nit

File: aten/src/ATen/native/Resize.cpp (hunk 1 above)
Comment (jansel) [reply aten/src/ATen/native/Resize.cpp:332]: This is just moving code.  Not new code.

You have a const ptr to the wrapper class, not the StorageImpl.

**Review by yf225 — APPROVED** [review]

**Pull request closed without merging**